plugins/processors/gpuattributes/internal/awsneuron_metric_modifier.go

```diff
@@ -286,6 +305,39 @@ func (md *AwsNeuronMetricModifier) duplicateMetrics(metricsSlice pmetric.MetricS
 	}
 }
 
+func (md *AwsNeuronMetricModifier) aggregateCoreUtilizationMetrics(originalMetric pmetric.Metric) pmetric.MetricSlice {
+	newMetricSlice := pmetric.NewMetricSlice()
+	originalMetricDatapoints := originalMetric.Sum().DataPoints()
+	aggregatedValuesPerCore := map[NeuronCoreUtilizationDatapointAggregationKey]float64{}
+	for i := 0; i < originalMetricDatapoints.Len(); i++ {
+		originalDatapoint := originalMetricDatapoints.At(i)
+		runtimeTag, _ := originalDatapoint.Attributes().Get(RuntimeTag)
+		coreIDTag, _ := originalDatapoint.Attributes().Get(NeuronCoreLabel)
+		key := NeuronCoreUtilizationDatapointAggregationKey{runtimeTag: runtimeTag.Str(), coreID: coreIDTag.Str()}
+		aggregatedValuesPerCore[key] = max(aggregatedValuesPerCore[key], originalDatapoint.DoubleValue(), 0)
+	}
+
+	if len(aggregatedValuesPerCore) == 0 {
+		return newMetricSlice
+	}
+
+	aggregatedMetric := setMetricMetadata(newMetricSlice.AppendEmpty(), originalMetric.Name(), originalMetric.Unit())
+	aggregateDatapoints := aggregatedMetric.SetEmptySum().DataPoints()
+	firstOriginalDatapoint := originalMetricDatapoints.At(0)
+	// Creating body for the aggregated metric and add it to the new newMetricSlice for each Core
+	for aggregatedMetricMetadata, value := range aggregatedValuesPerCore {
+		datapoint := aggregateDatapoints.AppendEmpty()
+		firstOriginalDatapoint.CopyTo(datapoint)
+		datapoint.SetDoubleValue(value)
+		datapoint.Attributes().PutStr(RuntimeTag, aggregatedMetricMetadata.runtimeTag)
+		datapoint.Attributes().PutStr(NeuronCoreLabel, aggregatedMetricMetadata.coreID)
+		datapoint.Attributes().PutStr(NeuronCoreAttributeKey, "core"+aggregatedMetricMetadata.coreID)
+		coreId, _ := strconv.Atoi(aggregatedMetricMetadata.coreID)
```

Comment:
```suggestion
		coreID, _ := strconv.Atoi(aggregatedMetricMetadata.coreID)
```
Looks like you need to fix this too